pkg/api/types.go

```diff
@@ -195,6 +195,32 @@ type VolumeSource struct {
 	NFS *NFSVolumeSource `json:"nfs"`
 }
 
+// PersistentVolumeSource is similar to VolumeSource but meant for the administrator who creates PVs.
+// Exactly one of its members must be set.
+type PersistentVolumeSource struct {
+	// GCEPersistentDisk represents a GCE Disk resource that is attached to a
+	// kubelet's host machine and then exposed to the pod.
+	GCEPersistentDisk *GCEPersistentDiskVolumeSource `json:"persistentDisk"`
```

Comment:
This PR is the smallest implementation of PersistentVolumePlugin I could devise.  It requires only that AccessModeTypes be defined.

All other API changes are in #5800.